File: Python/Samples/Mediator/MediatorWeight.py
from Mediator import Mediator
from OptionColleague import OptionColleague
from CreditColleague import CreditColleague
from LoanColleague import LoanColleague

class MediatorWeight(Mediator):

    def __init__(self, weightOption, weightCredit, weightLoan):
        self.option = OptionColleague()
        self.credit = CreditColleague()
        self.loan = LoanColleague()

        self.weightOption = weightOption
        self.weightCredit = weightCredit
        self.weightLoan = weightLoan

    def score(self):
        scoreOption = self.option.score() * self.weightOption
        scoreCredit = self.credit.score() * self.weightCredit
        scoreLoan = self.loan.score() * self.weightLoan

        return scoreOption + scoreCredit + scoreLoan

File: CSharp/DP.Domain/Samples/Mediator/MediatorWeight.cs
using System.Diagnostics;

namespace DP.Domain.Samples.Mediator
{
    public class MediatorWeight : IMediator
    {
        public IColleague Option {get;set;}
        public IColleague Credit {get;set;}
        public IColleague Loan {get;set;}

        private decimal _weightOption = 0;
        private decimal _weightCredit = 0;
        private decimal _weightLoan = 0;

        public MediatorWeight(decimal weightOption, decimal weightCredit, decimal weightLoan)
        {
            this._weightOption = weightOption;
            this._weightCredit = weightCredit;
            this._weightLoan = weightLoan;

            this.Option = new OptionColleague();
            this.Credit = new CreditColleague();
            this.Loan = new LoanColleague();
        }
        public decimal Score()
        {
            decimal scoreOption = this.Option.Score()* this._weightOption;
            decimal scoreCredit = this.Credit.Score()* this._weightCredit;
            decimal scoreLoan = this.Loan.Score()* this._weightLoan;

            return scoreOption + scoreCredit + scoreLoan;
        }
    }
}